### DISCUSSION TITLE
onColumnFiltersChange

### DISCUSSION THREAD

**Discussion by fjdsoft** [KevinVandy/material-react-table#449] [Q&A]
How can i use onColumnFiltersChange in a react class component.

I try.

onColumnFiltersChange={this._setColumnFilters}

_setColumnFilters = (filters) => this.setState({columnFilters: filters});

but dont work, this.state.columnFilters not get de values.

thanks.

**Comment by KevinVandy** [comment]
I personally have not used MRT in class components, but I think you will have to take into account that the on[State]Change props will have either a raw value in their updater, or a callback function expecting previous state. There is an example in the state management docs for how to write a function handler for this.
https://www.material-react-table.com/docs/guides/table-state-management#add-side-effects-in-set-state-callbacks

Probably something like this?

```tsx
_setColumnFilters = (updater: MRT_Updater) => {
  this.setState(updater instanceof Function ? updater(prevPagination) : updater)
};

render() {
  return (
    <MaterialReactTable
      columns={columns}
      data={data}
      state={{ pagination }}
      onColumnFiltersChange={_setColumnFilters}
    />
  );
}
